mapping_collection not defined

```
/work/00434/eijkhout/dealii/dealii-9.4.1/include/deal.II/numerics/vector_tools_project.templates.h:1020:49: warning: instantiation of variable 'dealii::hp::StaticMappingQ1<1>::mapping_collection'
required here, but no definition is available [-Wundefined-var-template]
    project(hp::StaticMappingQ1<dim, spacedim>::mapping_collection,
                                                ^
/work/00434/eijkhout/dealii/dealii-9.4.1/include/deal.II/hp/mapping_collection.h:150:45: note: forward declaration of template entity is here
    static MappingCollection<dim, spacedim> mapping_collection;
```

Logs attached.
[deal94.zip](https://github.com/dealii/dealii/files/10438367/deal94.zip)

--- a/include/deal.II/hp/mapping_collection.h
+++ b/include/deal.II/hp/mapping_collection.h
@@ -152,6 +152,7 @@ namespace hp
   };
 
 
+
   /* --------------- inline functions ------------------- */
 
   template <int dim, int spacedim>
@@ -179,6 +180,19 @@ namespace hp
   MappingCollection<dim, spacedim>
     StaticMappingQ1<dim, spacedim>::mapping_collection =
       MappingCollection<dim, spacedim>(MappingQ1<dim, spacedim>{});
+
+
+#ifndef DOXYGEN
+  // Declare the existence of explicit instantiations of the class
+  // above to avoid certain warnings issues by clang and
+  // newer (LLVM-based) Intel compilers:
+  extern template struct StaticMappingQ1<1, 1>;
+  extern template struct StaticMappingQ1<1, 2>;
+  extern template struct StaticMappingQ1<1, 3>;
+  extern template struct StaticMappingQ1<2, 2>;
+  extern template struct StaticMappingQ1<2, 3>;
+  extern template struct StaticMappingQ1<3, 3>;
+#endif
 
 } // namespace hp
 

